### PR TITLE
Implement GORM/Migration System

### PR DESCRIPTION
**Pull request by ilanamt**

### CODE REVIEW

**Review by rynmrtn — COMMENTED** [review]
please include:
- docker-compose
- db/main.go
- initdb.sh

File: db/testing/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+services:
+  postgres:
+    container_name: postgres_demo
+    image: postgres:latest
+    restart: always
+    environment:
+      POSTGRES_USER: myuser
+      POSTGRES_PASSWORD: mypass
+      POSTGRES_DB: test
```
Comment (rynmrtn) [inline db/testing/docker-compose.yml:12]: let's rename this asset_db

File: db/testing/createdb.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+CREATE TABLE IF NOT EXISTS test(id int);
```
Comment (rynmrtn) [inline db/testing/createdb.sql:1]:
```suggestion
CREATE TABLE IF NOT EXISTS asset_db(id int);
```

File: db/testing/docker-compose.yml (hunk 1 above)
Comment (rynmrtn) [inline db/testing/docker-compose.yml:12]:
```suggestion
      POSTGRES_DB: asset_db
```

File: db/main.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+func main() {
+	migrations := &migrate.FileMigrationSource{
+		Dir: "migrations/postgres",
+	}
+
+	dsn := "host=localhost dbname=test user=myuser password=mypass sslmode=disable"
```
Comment (rynmrtn) [inline db/main.go:16]:
```suggestion
	dsn := "host=localhost dbname=asset_db user=myuser password=mypass sslmode=disable"
```

File: db/main.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+
+	migrate "github.com/rubenv/sql-migrate"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
```
Comment (rynmrtn) [inline db/main.go:10]:
```suggestion
// main is a placeholder while additional commands are being developed
// TODO: this should be removed when the db cli command supports init, migrate, and 
// other related commands
```

**Review by fnmendez — APPROVED** [review]
lgtm

File: db/testing/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+version: '3'
```
Comment (rynmrtn) [inline db/testing/docker-compose.yml:1]: please move this to the repo root directory

File: db/testing/initdb.sh [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
```
Comment (rynmrtn) [inline db/testing/initdb.sh:1]: please move this to the resources/ folder and update docker-compose w/ the correct path

**Review by rynmrtn — COMMENTED** [review]
couple of small tweaks @ilanamt. 

Also, I changed the Pull request to be a PR into a `postgres-prototype` branch instead of `master`

File: db/testing/initdb.sh (hunk 6 above)
Comment (ilanamt) [reply db/testing/initdb.sh]: done!

File: db/testing/docker-compose.yml (hunk 5 above)
Comment (ilanamt) [reply db/testing/docker-compose.yml]: done!

**Review by rynmrtn — APPROVED** [review]

**Review by rodrigopv — APPROVED** [review]

**Pull request merged**